app/lib/actions/pre_upgrade_action.rb

```diff
@@ -0,0 +1,33 @@
+module Actions
+  module ForemanLeapp
+    class PreUpgradeAction < Actions::EntryAction
+      def self.subscribe
+        Actions::RemoteExecution::RunHostJob
+      end
+
+      def plan(job_invocation, host, *args)
+        return unless job_invocation.job_category == ::ForemanLeapp::JOB_CATEGORY
+
+        plan_self(trigger_success: triggering_action.output[:success], host_id: host.id)
```

Comment:
Will the trigger_success be ever used for something or is it used only to order the execution of the actions?

Reply:
It's there just for the order of execution.

Reply:
That means you can drop it. Originally I suggested this to order the steps in run phase, but because the logic was moved into finalize and actions in finalize are run in sequence in the order in which they were planned, it is not needed anymore